serpentTools/parsers/fissionMatrix.py

```diff
@@ -1,15 +1,326 @@
+        title: str
+            Plot title
+        xlabel: str
+            x-axis label
+        ylabel: str
+            y-axis label
+        cmap: str
+            Color map
+        logScale: bool
+            Log-scale on, True; Log-scale off, False
+        cbarLabel: None or str
+            Label to apply to colorbar
+
+        """
+        self._matPlot(self.fMatU, ax, title, xlabel, ylabel, cmap, logScale,
+                      cbarLabel)
+
+    def _matPlot(self, A, ax=None, title=None, xlabel=None, ylabel=None, cmap=None,
+                 logScale=False, cbarLabel=None):
+        v = A.shape
+        xticks = range(0, v[0])
+        ax = cartMeshPlot(A, xticks, xticks[::-1], ax=ax, logScale=logScale,
+                          cbarLabel=cbarLabel)
+        formatPlot(ax, title=title, xlabel=xlabel, ylabel=ylabel, cmap=cmap)
+        plt.xticks([])
+        plt.yticks([])
+
+    def fMatEig(self):
+        """
+        Computes the Fission Matrix Eigenpairs
+
+        Returns
+        -------
+            self.domEigVal: float
+                Dominant Eigenvalue
+            self.domEigVec: np.array
+                Dominant Eigenvector
+
+        """
+        eigValVec, eigVecMat = np.linalg.eig(self.fMat)
+        self.eigVecMat = self.eigVecNorm(eigVecMat)
+        self.domEigVal = eigValVec[0]
+        self.eigValVec = eigValVec
+        self.domEigVec = self.eigVecMat[:, 0]
+        self.domRatio = self.eigValVec[1] / self.eigValVec[0]
+        return self.domEigVal, self.domEigVec
+
+    def eigVecNorm(self, w):
+        # Flips and normalizes eigenvectors
+        z = w[:, 0]
+        lunghezza = len(z)
+        cont = len(z[z < 0])
+        if cont == lunghezza or cont == 0:
+            w = w / np.sum(w[:, 0])
+        else:
+            error('The Dominant Eigenvector is not Positive')
+        return w
+
+    def eigVecPlot(self, eigNum=1, xdata=None, ax=None,
+                   title='Neutron fission source', xlabel=None, ylabel=None,
+                   grid=True):
+        """
+        Plots the spatial distribution of the eigNum-th mode
+
+        Parameters
+        ----------
+        eigNum: list
+            Mode number (eigNum>0)
+        xdata: np.array
+            x axis data
+        ax: object
+            Axis object
+        title: str
+            Plot title
+        xlabel: str
+            x-axis label
+        ylabel: str
+            y-axis label
+        grid: bool
+            Grid (True), Not Grid (False)
+
+        Returns
+        -------
+        ax: object
+            Axis object
+
+        Raises
+        -------
+        ValueError
+            when eigNum is not consistent with the number of eigenvectors
+
+        """
+        # Check and initialization
+        lunghezza = len(self.domEigVec)
+        mex = 'The mode-number must be in the range 1-%s' % lunghezza
+        limCheck(eigNum, 1, lunghezza, mex)
+        if xdata is None:
+            xdata = range(0, len(self.domEigVec))
```

Comment:
Ah, so here is where the real restriction to 1D comes in to play....interesting.